mlir/include/mlir/IR/OperationSupport.h

```diff
@@ -1199,6 +1212,10 @@ class OpPrintingFlags {
   /// Elide printing large resources based on size of string.
   std::optional<uint64_t> resourceStringCharLimit;
 
+  /// Print large element attributes with hex strings if the number of elements
+  /// is larger than the upper limit.
+  int64_t elementsAttrHexElementLimit = 100;
```

Comment:
i'm in doubts whether this should be aligned to the rest of the int values (so, becoming `std::optional<>`). the difference is that e.g. element attr elision is "off" by default so `std::nullopt` is its natural state; but hex printing is **"on"** by default, and then it must be explicitly disabled (at least current behavior is like that and I'm not sure I should change it) - to do this, an explicit `-1` has to be provided, so both `std::nullopt` and `-1` would encode "disabled" state which seems too much for me.

otoh, I can hide the `std::optional<>` handling purely inside the functions so it's not too exposed "outside".

Reply:
LG